src/js/components/alert.js

```diff
@@ -5,43 +5,67 @@
 
 export default class Alert {
 
-  constructor(options) {
-    this.handleClick = this.handleClick.bind(this);
+  constructor(element) {
 
-    this.init(options);
-  }
-
-  init(options) {
-    document.getElementById(options.id).closest('[aria-labelledby]').querySelector('[data-alert-close]').addEventListener('click', () => { this.handleClick(options.id); }, false);
-  }
+    // Instance properties
+    this.element = element;
+    this.closeAttribute = ['data-alert-close'];
+    this.closeSelector = `[${this.closeAttribute}]`;
+    this.ariaAttribute = ['aria-labelledby'];
+    this.ariaSelector = `[${this.ariaAttribute}]`;
 
-  handleClick(id) {
-    const dismissButton = event.target.closest('[data-alert-close]');
+    // bind methods
+    this._handleClick = this._handleClick.bind(this);
 
-    // If the target wasn't the dismiss button bail.
-    if (!dismissButton) return;
+    // Check to make sure that a DOM element was passed in for initialization
+    if (!(Array.isArray(this.element))) {
+      throw new TypeError(
+        'A DOM element should be passed as the first argument to initialize the /alert'
+      );
+    }
 
-    this.dismissAlert(id);
+    this.init();
   }
 
-  dismissAlert(id, callback) {
-    let alert = document.querySelector('[aria-labelledby="' + id + '"]');
+  _handleClick(event) {
+    const dismissAria = event.target.closest(this.ariaSelector);
 
-    if (!alert) {
-      alert = document.getElementById(id);
+    // If the target wasn't the Aria element bail.
+    if (!dismissAria) {
+      throw new Error(
+        'Couldn\'t locate the Aria selector for the element.'
+      );
     }
 
+    this.dismissAlert(dismissAria);
+
+  }
+
+  dismissAlert(dismissAria) {
+    let alert = dismissAria;
+
     if (!alert) {
       throw new Error(
-        'Could not find an alert with the id of ' + id + ' to dismiss.'
+        'Could not find an alert attached to the close selector to dismiss.'
       );
     }
 
     alert.remove();
+  }
 
-    if (callback && typeof callback === 'function') {
-      callback();
-    }
+  init() {
+    // Add click handlers
+    this.element.forEach((item) => {
```

Comment:
We'll need to rethink this strategy for attaching event handlers a bit. It looks like you are trying to look for and attaching  event listeners to multiple close buttons here, but there should only ever be one close button per alert.

What you'll need to do is query the main alert element (`this.element`) for the close button and then only add an event listener to that single close button inside the main alert element.

So this should become:

```js
init() {
  let dismissButton = this.element.querySelector(this.closeSelector);
  dismissButton.addEventListener('click', this._handleClick, false);
}
```